src/graphics/shader-processor.js

```diff
@@ -57,7 +89,15 @@ class ShaderProcessor {
 
         // uniforms - merge vertex and fragment uniforms, and create shared uniform buffers
         const uniforms = vertexExtracted.uniforms.concat(fragmentExtracted.uniforms);
-        const uniformsData = ShaderProcessor.processUniforms(device, uniforms, shaderDefinition.processingOptions);
+
+        // parse uniform lines
+        /** @type {Array<UniformLine>} */
+        const parsedUniforms = [];
+        uniforms.forEach((line) => {
+            parsedUniforms.push(new UniformLine(line));
+        });
```

Comment:
There is no need for a type annotation when it can be inferred through mapping, which boils it all down to one line:

```suggestion
        const parsedUniforms = uniforms.map(line => new UniformLine(line));
```